receiver/dockerstatsreceiver/receiver.go

```diff
@@ -264,3 +265,9 @@ func (r *receiver) recordCPUMetrics(now pcommon.Timestamp, cpuStats *dtypes.CPUS
 		r.mb.RecordContainerCPUUsagePercpuDataPoint(now, int64(v), fmt.Sprintf("cpu%s", strconv.Itoa(coreNum)))
 	}
 }
+
+func (r *receiver) recordInspectMetrics(now pcommon.Timestamp, containerJSON *dtypes.ContainerJSON) {
+	r.mb.RecordContainerCPULimitDataPoint(now, float64(containerJSON.HostConfig.NanoCPUs)/1e9)
```

Comment:
Is this approach reliable every time? I'm just asking because I see that we do it differently at Datadog, by relying on multiple attributes. Perhaps on different systems, different properties are available. Take a look please: https://github.com/DataDog/datadog-agent/blob/ab4050c774a8f672fc49c29c4dc5a9a829e45752/pkg/util/containers/metrics/docker/collector_linux.go#L139-L163

Reply:
@gbbr Thanks for pointing that out. I'm currently having a close look at these calculations. Appreciate your comments.